### PR TITLE
Feat: create defendant endpoint

### PR DESCRIPTION
**Pull request by NicollyRamos**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Issue Number: #54 

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior? (Optional)

Enpoints without swagger documentation.

### CODE REVIEW

File: src/main/java/org/instituteatri/deep/controller/DefendantController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private final DefendantService defendantService;
+
+    @GetMapping
+    public ResponseEntity<List<DefendantDTO>> getAll(){
+        List<DefendantDTO> defendantDTOList = defendantService.getAll();
+        LOGGER.info("Request to get all defendants received");
+        return ResponseEntity.ok(defendantDTOList);
+    }
+
+    @GetMapping("/{id}")
```
Comment (rafael-dev2021) [inline src/main/java/org/instituteatri/deep/controller/DefendantController.java]: Specify the return type on this route, for example, “find/{id}”

File: src/main/java/org/instituteatri/deep/controller/DefendantController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    @PostMapping("/create")
+    @Transactional
+    public ResponseEntity<DefendantDTO> create(@RequestBody DefendantDTO defendant){
+        DefendantDTO defendantDTO = defendantService.save(defendant);
+        LOGGER.info("Request to save a new defendant received");
+        return ResponseEntity.status(HttpStatus.CREATED.value()).body(defendantDTO);
+    }
+
+    @DeleteMapping("/{id}")
```
Comment (rafael-dev2021) [inline src/main/java/org/instituteatri/deep/controller/DefendantController.java]: Specify the return type on this route, for example, “delete/{id}”

File: src/main/java/org/instituteatri/deep/service/DefendantService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public DefendantDTO getById(String id){
+        Defendant defendant = defendantRepository.findById(id)
+                .orElseThrow(() -> new BadRequestException("Defendant not found!"));
+        return modelMapper.map(defendant, DefendantDTO.class);
+    }
+
+    @Transactional
+    public DefendantDTO save(DefendantDTO defendantDTO){
```
Comment (rafael-dev2021) [inline src/main/java/org/instituteatri/deep/service/DefendantService.java]: Specify the name of the method more explicitly

File: src/main/java/org/instituteatri/deep/service/DefendantService.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        Defendant defendant = modelMapper.map(defendantDTO, Defendant.class);
+        Defendant newDefendant = defendantRepository.save(defendant);
+        LOGGER.info("Inserting {} to the database", defendant);
+        return modelMapper.map(newDefendant, DefendantDTO.class);
+    }
+
+    public void delete(String id){
```
Comment (rafael-dev2021) [inline src/main/java/org/instituteatri/deep/service/DefendantService.java]: Specify the name of the method more explicitly

**Review by rafael-dev2021 — CHANGES_REQUESTED** [review]

**Review by rafael-dev2021 — APPROVED** [review]
LGTM

**Review by Rpkevin — APPROVED** [review]
LGTM

**Review by ffx64 — APPROVED** [review]
LGTM

**Pull request merged**